internal/metric/mock.go

```diff
@@ -71,13 +77,19 @@ const (
 )
 
 func (i *Instrument) AcquireHandle(labels apimetric.LabelSet) apimetric.HandleImpl {
+	if ld, ok := labels.(apimetric.LabelSetDelegate); ok {
+		labels = ld.Delegate()
+	}
```

Comment:
This check should be done for sdk as well.

Reply:
Yes! This crossed my mind as I was walking out the office. Suggests more tests, too :)

Reply:
Done.